Userland/Libraries/LibSoftGPU/Device.h

```diff
@@ -85,14 +95,20 @@ class Device final {
     void blit_to(Gfx::Bitmap&);
     void wait_for_all_threads() const;
     void set_options(const RasterizerOptions&);
+    void set_light_model_params(const LightModelParameters&);
     RasterizerOptions options() const { return m_options; }
+    LightModelParameters light_model() const { return m_lighting_model; }
     Gfx::RGBA32 get_backbuffer_pixel(int x, int y);
     float get_depthbuffer_value(int x, int y);
 
     NonnullRefPtr<Image> create_image(ImageFormat, unsigned width, unsigned height, unsigned depth, unsigned levels, unsigned layers);
 
     void set_sampler_config(unsigned, SamplerConfig const&);
 
+    size_t number_of_lights() const { return m_lights.size(); }
+    Light& light(size_t index) { return m_lights.at(index); }
```

Comment:
Please have a look at `SoftwareGLContext::sync_device_config()` and set the light data in a similar fashion to `Device::set_sampler_config()`